### DISCUSSION TITLE
Trying to implement custom filter for tree

### DISCUSSION THREAD

**Discussion by cramatt** [zauberzeug/nicegui#1971] [Q&A]
I would like to filter a tree using an input, for which this is a great working example. https://github.com/zauberzeug/nicegui/discussions/1924#discussioncomment-7466148

However I would like to supply a custom filter function so that I can filter on node label and/ or id (label appears to be default). 

Docs are here: https://quasar.dev/vue-components/tree

I started here expecting this to log, but it didn't work:

```python
search = ui.input('Filter').props('filter-method="(e) => console.log(e)"')
```
Additionally the docs note "For best performance, reference it from your scope and do not define it inline" but not sure how to do this with the `props` api 🤔

**Comment by falkoschindler** [comment]
There is no "filter-method" in QInput. You probably want to set it for the tree instead.
And if you pass JavaScript via `props()`, you need to indicate a "dynamic prop" using a leading colon ":". Then the client will convert the string into a proper function. [Here](https://nicegui.io/documentation/table#custom_sorting_and_formatting) is an example for dynamic props.
